boards/arm/apollo4p_evb/apollo4p_evb_defconfig

```diff
@@ -3,3 +3,4 @@
 # Copyright (c) 2023 Antmicro <www.antmicro.com>
 
 CONFIG_SOC_SERIES_APOLLO4X=y
+CONFIG_PINCTRL=y
```

Comment:
this is not required (some boards still do this for historical reasons), drivers need to `select PINCTRL` instead.

Reply:
Removed `CONFIG_PINCTRL=y` from defconfig, [added `select PINCTRL` to the UART driver](https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/drivers/serial/Kconfig.pl011#L10) instead.